aeron-cluster/src/test/java/io/aeron/cluster/IngressAdapterTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2014-2024 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.aeron.cluster;
+
+import static org.mockito.Mockito.*;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.aeron.cluster.codecs.AdminRequestDecoder;
+import io.aeron.cluster.codecs.AdminRequestEncoder;
+import io.aeron.cluster.codecs.AdminRequestType;
+import io.aeron.cluster.codecs.ChallengeResponseDecoder;
+import io.aeron.cluster.codecs.ChallengeResponseEncoder;
+import io.aeron.cluster.codecs.MessageHeaderDecoder;
+import io.aeron.cluster.codecs.MessageHeaderEncoder;
+import io.aeron.cluster.codecs.SessionCloseRequestDecoder;
+import io.aeron.cluster.codecs.SessionCloseRequestEncoder;
+import io.aeron.cluster.codecs.SessionConnectRequestDecoder;
+import io.aeron.cluster.codecs.SessionConnectRequestEncoder;
+import io.aeron.cluster.codecs.SessionKeepAliveDecoder;
+import io.aeron.cluster.codecs.SessionKeepAliveEncoder;
+import io.aeron.cluster.codecs.SessionMessageHeaderDecoder;
+import io.aeron.cluster.codecs.SessionMessageHeaderEncoder;
+
+class IngressAdapterTest
+{
+    private IngressAdapter adapter;
+    private ConsensusModuleAgent consensusModuleAgent;
+    private ExpandableArrayBuffer buffer;
+
+    @BeforeEach
+    public void setup()
+    {
+        buffer = new ExpandableArrayBuffer();
+        consensusModuleAgent = mock(ConsensusModuleAgent.class);
+        adapter = new IngressAdapter(0, consensusModuleAgent);
+    }
+
+    @SuppressWarnings("unused") // name used for test display name
+    @ParameterizedTest(name = "{index} {0}")
+    @MethodSource
+    public void shouldDelegateToConsensusModuleAgent(
```

Comment:
Methods do not need to be public in a package private class.

Reply:
done